server/src/main/kotlin/org/jetbrains/teamcity/vault/server/VaultProjectConnectionProvider.kt

```diff
@@ -64,14 +64,18 @@ class VaultProjectConnectionProvider(private val descriptor: PluginDescriptor) :
                 if (namespace != "" && !namespace.matches(namespaceRegex.toRegex())) {
                     errors.add(InvalidProperty(VaultConstants.FeatureSettings.NAMESPACE, "Non-default namespace should match regex '$namespaceRegex'"))
                 }
-                if (it[VaultConstants.FeatureSettings.ENDPOINT].isNullOrBlank()) {
-                    errors.add(InvalidProperty(VaultConstants.FeatureSettings.ENDPOINT, "Should not be empty"))
-                }
-                if (it[VaultConstants.FeatureSettings.ROLE_ID].isNullOrBlank()) {
-                    errors.add(InvalidProperty(VaultConstants.FeatureSettings.ROLE_ID, "Should not be empty"))
-                }
-                if (it[VaultConstants.FeatureSettings.SECRET_ID].isNullOrBlank()) {
-                    errors.add(InvalidProperty(VaultConstants.FeatureSettings.SECRET_ID, "Should not be empty"))
+
+                val awsIamAuth = it[VaultConstants.FeatureSettings.AWS_IAM_AUTH] ?: "false"
```

Comment:
Please change to `?.toBoolean() ?: false` as it's used elsewhere (grep `toBoolean` on project)